mailpoet/lib/Subscription/Comment.php

```diff
@@ -31,7 +31,21 @@ public function extendLoggedInForm($field) {
   }
 
   public function extendLoggedOutForm() {
-    echo $this->getSubscriptionField();
+    $allowedHtml = [
+      'p' => [
+        'class' => true,
+      ],
+      'label' => [
+        'for' => true,
+      ],
+      'input' => [
+        'type' => true,
+        'id' => true,
+        'value' => true,
+        'name' => true,
+      ],
+    ];
+    echo wp_kses($this->getSubscriptionField(), $allowedHtml);
```

Comment:
I would like to not ignore, if we do not have to. `self::getSubscriptionField` is sanitized now, but what about tomorrow? I think this is the basic philosophy behind late escaping.

Reply:
I would be ok with ignoring here since the method is private and used for generating the output. It feels similar to creating a variable escaping it and then using it in output.

Reply:
I agree with @costasovo, now we're literally [escaping right after creating an escaped input](https://github.com/mailpoet/mailpoet/blob/60dff543ab55ebcb9ed2255deadb52f32cd242a6/mailpoet/lib/Subscription/Comment.php#L33-L67):

```php
  public function extendLoggedOutForm() {
    $allowedHtml = [
      'p' => [
        'class' => true,
      ],
      'label' => [
        'for' => true,
      ],
      'input' => [
        'type' => true,
        'id' => true,
        'value' => true,
        'name' => true,
      ],
    ];
    echo wp_kses($this->getSubscriptionField(), $allowedHtml);
  }


  private function getSubscriptionField() {
    $label = $this->settings->get(
      'subscribe.on_comment.label',
      WPFunctions::get()->__('Yes, please add me to your mailing list.', 'mailpoet')
    );


    return '<p class="comment-form-mailpoet">
      <label for="mailpoet_subscribe_on_comment">
        <input
          type="checkbox"
          id="mailpoet_subscribe_on_comment"
          value="1"
          name="mailpoet[subscribe_on_comment]"
        />&nbsp;' . esc_html($label) . '
      </label>
    </p>';
  }
```

Reply:
Maybe we could make the private method `echo` directly and then all problems solved? 😄

Reply:
Unfortunately this method is also used to return the field (when you are logged in, we hook in a filter instead of a hook, changing that would be a BC
![image](https://user-images.githubusercontent.com/6458412/161536735-65168ea8-5f11-4ed3-88ee-a8088f87dd85.png)

In the end: I am okay with `ignore`, will not die to fight this battle :D